scriptlets/install_checkbox_snaps.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# Installs Checkbox and exports the installed version on the DUT
+# to: CHECKBOX_VERSION
+
+if [[ "$#" != "4" ]]; then
+  echo "Usage: $(basename ${BASH_SOURCE[0]}) risk runtime_name frontend_name frontend_track"
+  exit 1
+fi
+
+echo "Installing Checkbox snaps"
+
+RISK=$1
+shift
+RUNTIME_NAME=$1
+shift
+FRONTEND_NAME=$1
+shift
+FRONTEND_TRACK=$1
+shift
+
+wait_for_snap_complete
+_run sudo snap install $RUNTIME_NAME --channel=latest/$RISK
+wait_for_snap_complete
```

Comment:
The approach I've followed for every `snap-install` so far is to use it with `--no-wait`, in combination with `wait_for_snap_changes` (not `wait_for_snap_complete`). This makes it more robust and immune to any interference from auto-updates or reboots.

Reply:
(I wonder what is the difference between `wait_for_snap_changes` and `wait_for_snap_complete`, I used the other one in the beginning but by name this looked like it was more correct)

Reply:
(I wonder what is the difference between `wait_for_snap_changes` and `wait_for_snap_complete`, I used the other one in the beginning but by name this looked like it was more correct)

Reply:
`wait_for_snap_complete` was the name of the inline function previously used in snap-testing jobs. There is also a `wait_for_snap_complete` scriptlet written some time ago that was a variation of that function but had never been used. I do not clearly recall why I chose to name the reinvented script differently -- I assume I wanted to reflect the `snap changes` command that it monitors. I acknowledge it is confusing but I claim to compensate for that by the fact that it seems to work very well :grin: